tests/gtests/internals/test_sdpa.cpp

```diff
@@ -0,0 +1,1910 @@
+        case mask_type::twoD: mask_ptr = &mask; break;
+    }
+
+    //auto sdpaf16_pd = sdpa::primitive_desc(eng, t.m_query.get_desc(),
+    //t.m_key.get_desc(), t.m_value.get_desc(), mask_ptr, scale_dt,
+    //t.m_output.get_desc(), invert_scale, p.head_num, p.with_causal_mask, t.sdpa_attr);
+    //auto sdpaf16_p = sdpa(sdpaf16_pd);
+
+    //print_mem(t.m_query_test, "query_quantized");
+    //print_mem(t.m_key_quantized, "key_quantized");
+    auto sdpas8_pd = sdpa::primitive_desc(eng, t.m_query_test.get_desc(),
+            p.with_key_transposed ? t.m_key_t_quantized.get_desc()
+                                  : t.m_key_quantized.get_desc(),
+            t.m_value_quantized.get_desc(), mask_ptr, scale_dt,
+            t.m_output_quantized.get_desc(), invert_scale, p.head_num,
+            p.mask == mask_type::causal, t.sdpa_attr_quantized,
+            t.sdpa_kq_attr_quantized, t.sdpa_vs_attr_quantized);
+    auto sdpas8_p = sdpa(sdpas8_pd);
+
+    std::unordered_map<int, memory> s8_args = {{{DNNL_ARG_QUERIES, t.m_query},
+            {DNNL_ARG_VALUES, t.m_value_quantized},
+            {DNNL_ARG_DST, t.m_output_quantized}}};
+
+    if (p.with_key_transposed) {
+        s8_args[DNNL_ARG_KEYS] = t.m_key_t_quantized;
+    } else {
+        s8_args[DNNL_ARG_KEYS] = t.m_key_quantized;
+    }
+    if (scale_dt != mdt::undef) { s8_args[DNNL_ARG_SCALE] = t.m_scale; }
+
+    if ((p.kdt != mdt::f16 || p.kdt != mdt::bf16)
+            && p.qtype != quantize_type::no_quantization) {
+        s8_args[DNNL_ARG_ATTR_SCALES | DNNL_ARG_KEYS] = t.m_key_scales;
+        s8_args[DNNL_ARG_ATTR_ZERO_POINTS | DNNL_ARG_KEYS] = t.m_key_zp;
+    }
+    if ((p.vdt != mdt::f16 || p.vdt != mdt::bf16)
+            && p.qtype != quantize_type::no_quantization) {
+        s8_args[DNNL_ARG_ATTR_SCALES | DNNL_ARG_VALUES] = t.m_value_scales;
+        s8_args[DNNL_ARG_ATTR_ZERO_POINTS | DNNL_ARG_VALUES] = t.m_value_zp;
+    }
+    if (mask_ptr) { s8_args[DNNL_ARG_ATTN_MASK] = t.m_mask; }
+
+    std::unordered_map<int, memory> f16_args
+            = {{DNNL_ARG_QUERIES, t.m_query}, {DNNL_ARG_KEYS, t.m_key},
+                    {DNNL_ARG_VALUES, t.m_value}, {DNNL_ARG_DST, t.m_output}};
+    if (scale_dt != mdt::undef) { f16_args[DNNL_ARG_SCALE] = t.m_scale; }
+    if (mask_ptr) { f16_args[DNNL_ARG_ATTN_MASK] = t.m_mask; }
+
+    //print_mem(t.m_key_scales, "key_scales");
+    //print_mem(t.m_key_zp, "key_zp");
+    auto loop_s8 = [&] { sdpas8_p.execute(strm, s8_args); };
+    loop_s8();
+    strm.wait();
+    //auto loop_f16 = [&] { sdpaf16_p.execute(strm, f16_args); };
+    //loop_f16();
+    //strm.wait();
+    prim_sdpa_quant(p, t, eng, strm, t.m_query,
+            p.with_key_transposed ? t.m_key_t_quantized : t.m_key_quantized,
+            t.m_key_scales, t.m_key_zp, scale_dt, t.m_scale, t.m_mask,
+            t.m_value_quantized, t.m_value_scales, t.m_value_zp, t.m_output,
+            invert_scale);
+    strm.wait();
+    //print_mem(t.m_key, "key");
```

Comment:
Isn't it better to keep the commented code under a macro like you have in the file, ex : "#if 0"

Reply:
Normally I inspect these values one tensor at a time because otherwise its too difficult to parse. I can remove these comments if you think they shouldn't be part of the tests.

Reply:
Its okay to keep them, but I was just thinking to not have them commented but have a macro for it for ease of usage; for example:
```suggestion
#define PRINT_KEYS 0
#if PRINT_KEYS
    print_mem(t.m_key, "key");
#endif 
```